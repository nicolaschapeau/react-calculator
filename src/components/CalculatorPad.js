import React from 'react'




class CalculatorPad extends React.Component {
    render () {
        return (
            <div className="calculator-pad-container">

                <div className="calculator-pad-item ans" onClick={() => this.props.onAns()}>
                    <p>ANS</p>
                </div>
                <div className="calculator-pad-item"></div>
                <div className="calculator-pad-item clear" onClick={() => this.props.onDelete()}>
                    <p>CE</p>
                </div>
                <div className="calculator-pad-item clear-all" onClick={() => this.props.onClear()}>
                    <p>C</p>
                </div>

                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('7')}>
                    <p>7</p>
                </div>
                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('8')}>
                    <p>8</p>
                </div>
                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('9')}>
                    <p>9</p>
                </div>
                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick('^')}>
                    <p>^</p>
                </div>

                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('4')}>
                    <p>4</p>
                </div>
                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('5')}>
                    <p>5</p>
                </div>
                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('6')}>
                    <p>6</p>
                </div>
                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick('*')}>
                    <p>*</p>
                </div>

                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('1')}>
                    <p>1</p>
                </div>
                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('2')}>
                    <p>2</p>
                </div>
                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('3')}>
                    <p>3</p>
                </div>
                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick('/')}>
                    <p>/</p>
                </div>

                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick('√(')}>
                    <p>√</p>
                </div>
                <div className="calculator-pad-item number" onClick={() => this.props.onPadClick('0')}>
                    <p>0</p>
                </div>
                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick('+')}>
                    <p>+</p>
                </div>
                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick('-')}>
                    <p>-</p>
                </div>

                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick('(')}>
                    <p>(</p>
                </div>
                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick(')')}>
                    <p>)</p>
                </div>
                <div className="calculator-pad-item symbol" onClick={() => this.props.onPadClick('.')}>
                    <p>.</p>
                </div>
                <div className="calculator-pad-item result" onClick={() => this.props.onSubmit()}>
                    <p>=</p>
                </div>

            </div>
        )
    }
}





export default CalculatorPad