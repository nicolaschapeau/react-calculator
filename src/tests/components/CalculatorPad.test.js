import React from 'react';
import { shallow } from 'enzyme';
import CalculatorPad from '../../components/CalculatorPad';
import toJSON from 'enzyme-to-json'



// Snapshot
test('Should render CalculatorPad correctly', () => {
    const wrapper = shallow(<CalculatorPad />)

    expect(toJSON(wrapper)).toMatchSnapshot()
})
