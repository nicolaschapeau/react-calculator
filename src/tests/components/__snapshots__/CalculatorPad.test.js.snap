// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render CalculatorPad correctly 1`] = `
<div
  className="calculator-pad-container"
>
  <div
    className="calculator-pad-item ans"
    onClick={[Function]}
  >
    <p>
      ANS
    </p>
  </div>
  <div
    className="calculator-pad-item"
  />
  <div
    className="calculator-pad-item clear"
    onClick={[Function]}
  >
    <p>
      CE
    </p>
  </div>
  <div
    className="calculator-pad-item clear-all"
    onClick={[Function]}
  >
    <p>
      C
    </p>
  </div>
  <div
    className="calculator-pad-item number simulate"
    onClick={[Function]}
  >
    <p>
      7
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      8
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      9
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      ^
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      4
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      5
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      6
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      *
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      1
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      2
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      3
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      /
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      âˆš
    </p>
  </div>
  <div
    className="calculator-pad-item number"
    onClick={[Function]}
  >
    <p>
      0
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      +
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      -
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      (
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      )
    </p>
  </div>
  <div
    className="calculator-pad-item symbol"
    onClick={[Function]}
  >
    <p>
      .
    </p>
  </div>
  <div
    className="calculator-pad-item result"
    onClick={[Function]}
  >
    <p>
      =
    </p>
  </div>
</div>
`;
